core/src/main/scala/zio/jdbc/SqlFragment.scala

```diff
@@ -215,6 +222,8 @@ sealed trait SqlFragment { self =>
                                         val rowsUpdated = ps.executeLargeUpdate()
                                         val updatedKeys = ps.getGeneratedKeys
                                         (rowsUpdated, ZResultSet(updatedKeys))
+                                      }.refineOrDie { case e: SQLException =>
```

Comment:
Change: I believe connection timeouts are not reported as `SQLException`, and I think reporting them via `die` is a regression in the API. Given it's probably driver specific, we need to catch `Throwable` and use another case class to report it (i.e. we probably cannot use `ZSQLException` in the return type). 🤔

Reply:
If I am not mistaken `executeLargeUpdate()` can throw a `SQLTimeoutException`, but this should be caught as `SQLTimeoutException` is a subclass of `SQLException`. https://docs.oracle.com/javase/8/docs/api/java/sql/PreparedStatement.html#executeLargeUpdate--

Or do you mean that the driver would report the `SQLTimeoutException` using a different class?

Reply:
yeah, you're right, my bad (i've misinterpreted a few issues reported in our Sentry) 🤦

Reply:
But I think that it would be very interesting to specify retry policies in an easy manner. Maybe for this it should be better to encode `SQLTimeoutException` in a different case class. Or maybe this complicates further the API.